### ISSUE TITLE
builder/qemu: Support on-board NICs

### ISSUE THREAD

**Issue by radeksimko** [hashicorp/packer#7524]
As mentioned in https://www.qemu.org/2018/05/31/nic-parameter/ the `-device` & `-netdev` flags are unfortunately not equivalent to `-nic` or (deprecated) `-net`.

> The `-device` option can only be used for pluggable NICs. Boards (e.g. embedded boards) which feature an on-board NIC cannot be configured with `-device` yet.

Because Packer hard-codes `-device` and `-netdev` flags

https://github.com/hashicorp/packer/blob/29e608e50508d8f142d5e43ff0ef686ff9014f95/builder/qemu/step_run.go#L116

https://github.com/hashicorp/packer/blob/29e608e50508d8f142d5e43ff0ef686ff9014f95/builder/qemu/step_run.go#L81-L86

it's much more difficult to setup network between host/guest with on-board NIC (such as `smc91c111` used in Raspberry Pi). It might work if one creates some virtual adapters manually (I haven't tried myself), but it's far from trivial.

### Proposal 

Introduce a way to execute Packer without passing any `-device` or `-netdev` flags to QEMU.

Basically I'd like it to run

```sh
$ qemu-system-arm
...
	-nic user,id=user.0,hostfwd=tcp::<random-ssh-port>-:22
```
instead of the following
```sh
$ qemu-system-arm
...
	-device smc91c111,netdev=user.0 \
	-netdev user,id=user.0,hostfwd=tcp::<random-ssh-port>-:22
```
```
qemu-system-arm: -device smc91c111,netdev=user.0: Parameter 'driver' expects pluggable device type
```

### Possibly related

https://github.com/hashicorp/packer/issues/6947

**Comment by kondor6c** [comment]
Could #7437 also be related?

**Comment by radeksimko** [comment]
@kondor6c I don't believe it's related to this issue, except that we might also need a similar list of allowed devices (for all QEMU versions and architectures/binaries) for `-nic` if we make it a 1st class parameter in Packer.

I was more thinking along the lines of making `-device` and `-netdev` optional, which would at least make it possible to pass `-nic` like any other extra argument through [`qemuargs`](https://www.packer.io/docs/builders/qemu.html#qemuargs). In that case we wouldn't do any validation.
